src/bento/components/bento-mega-menu/1.0/component/Content.tsx

```diff
@@ -0,0 +1,55 @@
+import objStr from 'obj-str';
+
+import * as Preact from '#preact';
+import {useLayoutEffect} from '#preact';
+import {ComponentChildren} from '#preact/types';
+import {propName} from '#preact/utils';
+
+import {useMegaMenuItem} from './Item';
+import {AriaAttributes, AsComponent, AsProps} from './types';
+
+import {useStyles} from '../component.jss';
+
+type ContentProps<TAs extends AsComponent> = AsProps<TAs> & {
+  children?: ComponentChildren;
+  class?: string;
+  id?: string;
+};
+
+export function Content<TAs extends AsComponent = 'div'>({
+  as: As = 'div',
+  children,
+  id: idProp,
+  [propName('class')]: className,
+  ...rest
+}: ContentProps<TAs>) {
+  const classes = useStyles();
+  const {actions, isOpen, itemId} = useMegaMenuItem();
+
+  // If this ID is set, pass it to the parent:
+  useLayoutEffect(() => {
+    if (idProp) {
+      actions.overrideItemId(idProp);
+    }
+  }, [actions, idProp]);
+
+  const ariaAttrs: AriaAttributes = {
+    'aria-modal': isOpen,
+  };
+
+  return (
+    <As
+      role="dialog"
+      id={itemId}
+      {...ariaAttrs}
+      {...rest}
+      class={objStr({
+        [className!]: !!className,
```

Comment:
Interesting! In other places I have just seen `[className]: true`. Is this so we avoid "undefined" as a class name if it doesn't exist?

Reply:
Yes, that's why I did this, I didn't want `undefined` classes and thought this approach was the cleanest.
I just wish we used `clsx` which is so much simpler, but in one PR I implemented it (as a 1-liner, no less) but I got pushback to use `objStr` instead ... hence this workaround.